src/datasets.rs

```diff
@@ -133,30 +144,51 @@ impl Datasets {
     }
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+fn netdata_from_mapping<'a>(
+    dref: Option<DatasetRef>,
+    default_name: &'a str,
+    mapping: &'a DatasetsMapping,
+) -> (Option<DatasetId>, Option<&'a NetDataset>) {
+    let (tmp, key) = match &dref {
+        Some(DatasetRef::Id(id)) => (Some(id.clone()), default_name),
```

Comment:
You're already accepting `dref` by value, so it should be possible to avoid cloning
```suggestion
    let (tmp, key) = match dref {
        Some(DatasetRef::Id(id)) => (Some(id), default_name),
```